### PR TITLE
[JENKINS-46719] Check if container exists in pod at all

### PR DESCRIPTION
**Pull request by marvinthepa**
https://issues.jenkins-ci.org/browse/JENKINS-46719

### CODE REVIEW

File: src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java [hunk 1]
```diff
@@ -244,23 +244,22 @@ public void kill(Map<String, String> modelEnvVars) throws IOException, Interrupt
...
-                return false;
-            }
-
             private void waitUntilContainerIsReady() throws IOException {
                 try {
-                    client.pods().inNamespace(namespace).withName(podName)
+                    Pod pod = client.pods().inNamespace(namespace).withName(podName)
                             .waitUntilReady(CONTAINER_READY_TIMEOUT, TimeUnit.MINUTES);
+
+                    for (ContainerStatus info : pod.getStatus().getContainerStatuses()) {
```
Comment (carlossg) [inline src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java]: there was a check for `pod.getStatus() == null || pod.getStatus().getContainerStatuses() == null` IIRC null was possible during startup

Comment (marvinthepa) [reply src/main/java/org/csanchez/jenkins/plugins/kubernetes/pipeline/ContainerExecDecorator.java]: I thought that `waitUntilReady` should take care of that, but I'll re-add it to make sure.

**Comment by marvinthepa** [comment]
> I haven't checked but I guess `waitUntilReady` will return when one of the containers errors?

Unfortunately, it doesn't, but in that situation we will rather get into the #196 situation in that the pipeline will block at `node` until we run into the timeout.

This pull request is mainly about the situation where a user accidentally tries to execute a command in a container that is not even specified in the current node - they shouldn't have to wait for the timeout.

**Comment by marvinthepa** [comment]
N.B. maybe the switch on `info.getReady()` does not make sense (because `waitUntilReady` should actually only return `ready` containers), but I guess it does not hurt, either.

**Pull request merged**